utils/wasm-builder/src/optimize.rs

```diff
@@ -123,3 +129,22 @@ pub fn do_optimization(
     }
     Ok(())
 }
+
+pub fn check_exports(module: &Module, path: &Path) -> Result<()> {
+    if module
+        .export_section()
+        .ok_or_else(|| {
+            crate::builder_error::BuilderError::ExportSectionNotFound(path.to_path_buf())
+        })?
+        .entries()
+        .iter()
+        .any(|entry| {
+            matches!(entry.field(), "init" | "handle")
+                && matches!(entry.internal(), Internal::Function(_))
```

Comment:
As it is a lazy-AND we are to move more cheap expression in the first place I guess.
```suggestion
            matches!(entry.internal(), Internal::Function(_))
                && matches!(entry.field(), "init" | "handle")
```